box-node-custom-skills-sample/index.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @fileoverview Main sample skill lambda handler.
+ */
+
+/**
+ * Module dependencies
+ */
+const ErrorParser = require('error-parser');
+const MetadataCardAdaptor = require('metadata-card-adaptor');
+const { SkillsStatusEnum, SkillsErrorEnum } = require('skills-kit');
+
+const CUSTOM_SKILL_NAME = 'box-custom-sample-skill-node';
+
+/**
+ * This is the main function that the Lambda will call when invoked.
+ *
+ * @param {Object} event - data from the event, including the payload of the webhook, that triggered this function call
+ * @param {Object} context - additional context information from the request (unused in this example)
+ * @param {Function} callback - the function to call back to once finished
+ * @return {null} - lambda callback
+ */
+module.exports.handler = async (event: Object, context: Object, callback: Function): void => {
+    // log the box skills event recieved from box event pump, to see what data is availabe to you regarding a file on which your skill is invoked.
+    Logger.logDebug(`Box Event received: ${JSON.stringify(event)}`);
+
+    let skillInvocationsClient;
+    try {
+        // parse the incoming event to it's individual compoenents, and get fileReadClient and skillInvocations for use within your skills code
+        const { body } = event;
+        const { id: boxRequestId, skill, source, token } = JSON.parse(body);
+        const { read: { access_token: boxFileReadToken }, write: { access_token: boxFileWriteToken } } = token;
+        const { fileReadClient, skillInvocations } = new SkillsKit(
+            boxRequestId,
+            skill.id,
+            CUSTOM_SKILL_NAME,
+            source.id,
+            boxFileReadToken,
+            boxFileWriteToken
+        );
+        skillInvocationsClient = skillComponents.skillInvocations;
+
+        // write back an intitial File Metadata card that event processing has started, to show on the file preview UI sidebar.
+        await skillInvocationsClient.savePending();
+
+        // @TODO use the file infromation from the incoming skills request to download the file from box, in one of two ways-
```

Comment:
As this is a sample, May we write the code and just giving the File id as configurable.

Reply:
i changed @TODO to <PLEASE WRITE YOUR OWN CODE HERE> to clearly indicate the Skill developer needs to write their own code here